src/components/Message/MessageLivestream.js

```diff
@@ -290,265 +395,129 @@ class MessageLivestream extends PureComponent {
-                      />
-                    </svg>
-                    {/* TODO: Translate following sentence */}
-                    <strong>/{message.command}</strong> is not a valid command
-                  </p>
-                )}
-                {message && message.status === 'failed' && (
-                  <p
-                    onClick={() => {
-                      if (message.status === 'failed' && handleRetry) {
-                        // FIXME: type checking fails here because in the case of a failed message,
-                        // `message` is of type Client.Message (i.e. request object)
-                        // instead of Client.MessageResponse (i.e. server response object)
-                        // @ts-ignore
-                        handleRetry(message);
-                      }
-                    }}
-                  >
-                    <svg
-                      width="14"
-                      height="14"
-                      xmlns="http://www.w3.org/2000/svg"
-                    >
-                      <path
-                        d="M7 0a7 7 0 1 0 0 14A7 7 0 0 0 7 0zm.875 10.938a.438.438 0 0 1-.438.437h-.875a.438.438 0 0 1-.437-.438v-.874c0-.242.196-.438.438-.438h.875c.241 0 .437.196.437.438v.874zm0-2.626a.438.438 0 0 1-.438.438h-.875a.438.438 0 0 1-.437-.438v-5.25c0-.241.196-.437.438-.437h.875c.241 0 .437.196.437.438v5.25z"
-                        fill="#EA152F"
-                        fillRule="evenodd"
-                      />
-                    </svg>
-                    {t && t('Message failed. Click to try again.')}
-                  </p>
-                )}
-              </div>
-
-              {hasAttachment &&
-                Attachment &&
-                attachments.map(
-                  /** @type {(item: import('stream-chat').Attachment) => React.ReactElement | null} Typescript syntax */
-                  (attachment, index) => (
-                    <Attachment
-                      key={`${message?.id}-${index}`}
-                      attachment={attachment}
-                      actionHandler={handleAction}
-                    />
-                  ),
-                )}
-
-              {galleryImages.length !== 0 && <Gallery images={galleryImages} />}
-
-              {message && (
-                <SimpleReactionsList
-                  reaction_counts={message.reaction_counts}
-                  reactions={message.latest_reactions}
-                  handleReaction={handleReaction}
-                />
-              )}
-
-              {!initialMessage && message && (
-                <MessageRepliesCountButton
-                  onClick={handleOpenThread}
-                  reply_count={message.reply_count}
-                />
-              )}
-            </div>
-          </div>
-        </div>
-      </React.Fragment>
-    );
-  }
-}
-
-export default withTranslationContext(MessageLivestream);
+      )) ||
+    null
```

Comment:
an early null return is a bit more cleaner than this logical or after a couple of ands

Reply:
Refactored: https://github.com/GetStream/stream-chat-react/pull/380/commits/932f4f66fb04df7bc00d7a517a42a168c07f9687